x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/QueryList.java

```diff
@@ -129,14 +129,13 @@ private IntFunction<Object> blockToJavaObject() {
                     BytesRefBlock bytesRefBlock = (BytesRefBlock) block;
                     if (inputDataType == IP) {
                         yield offset -> {
-                            bytesRefBlock.getBytesRef(offset, scratch);
-                            if (ipBytes.length != scratch.length) {
+                            final var bytes = bytesRefBlock.getBytesRef(offset, scratch);
```

Comment:
I thought scratch was used as mutable state?

Reply:
Yes, but it's optional. We don't always need to write the returned result to this argument.